### PR TITLE
Turn MISO docker container into docker-compose containers

### PR DESCRIPTION
**Pull request by morgantaschuk**
Convert Docker container to multiple with Docker compose, added multi-stage build, updated documentation

### CODE REVIEW

File: docs/_posts/2016-01-12-developers-manual.md [hunk 1]
```diff
@@ -408,16 +408,22 @@ Pull the tag or snapshot that you want to build and package it:
...
+    docker build --target webapp -t "misolims/miso-lims-webapp:${version}" --no-cache .
+    docker tag "misolims/miso-lims-migration:${version}" misolims/miso-lims-migration:latest
+    docker tag "misolims/miso-lims-webapp:${version}" misolims/miso-lims-webapp:latest
 
-Once the build completes, test it by launching it:
+Once the build completes, test it by launching it. This command uses the default 
+environment variables in .env and relies on `.miso_db_password` file existing.
 
-    docker run -p 8090:8080 --name "miso${version}" -t "misolims/miso-lims:${version}"
+    export MISO_TAG="0.2.35-SNAPSHOT" && docker-compose up
```
Comment (djcooke) [inline docs/_posts/2016-01-12-developers-manual.md]: Should MISO_TAG be `${version}`?

File: docs/_posts/2016-01-12-developers-manual.md [hunk 2]
```diff
@@ -408,16 +408,22 @@ Pull the tag or snapshot that you want to build and package it:
 
     export version="0.2.35-SNAPSHOT"
     git checkout "tags/${version}"
-    mvn -P external clean install package
-    docker build -t "misolims/miso-lims:${version}" --build-arg version="${version}" --no-cache .
+    docker build --target flyway-migrate -t "misolims/miso-lims-migration:${version}" --no-cache .
```
Comment (djcooke) [inline docs/_posts/2016-01-12-developers-manual.md]: target should be `flyway-migration`

File: docs/_posts/2016-01-12-developers-manual.md (hunk 1 above)
Comment (morgantaschuk) [reply docs/_posts/2016-01-12-developers-manual.md]: I had it as version but it's actually the tag on the docker container. As long as we're happy keeping the tags and versions in sync, I can substitute it.

Comment (morgantaschuk) [reply docs/_posts/2016-01-12-developers-manual.md]: Oh never mind, you mean the actual value. I can change that.

File: docs/_posts/2016-01-12-developers-manual.md (hunk 2 above)
Comment (morgantaschuk) [reply docs/_posts/2016-01-12-developers-manual.md]: Fixed

File: docs/_posts/2016-01-12-developers-manual.md (hunk 1 above)
Comment (morgantaschuk) [reply docs/_posts/2016-01-12-developers-manual.md]: Fixed

File: Dockerfile [hunk 3]
```diff
@@ -1,20 +1,89 @@
...
+#######################################################
+## Flyway database migration
+#######################################################
+FROM openjdk:8-jre-alpine as flyway-migration
+
+# set up flyway's dependencies and MISO's jar
+ARG FLYWAY_VERSION=3.2.1
+RUN apk --no-cache add --update bash openssl unzip
+
+RUN wget https://raw.githubusercontent.com/vishnubob/wait-for-it/master/wait-for-it.sh -O /wait-for-it.sh
```
[the remark this answers is not in the document]
Comment (morgantaschuk) [reply Dockerfile]: Fixed

**Pull request merged**